pkg/controller/deployment/deployment_controller.go

```diff
@@ -573,6 +578,23 @@ func (dc *DeploymentController) syncRecreateDeployment(deployment *extensions.De
 	return dc.syncDeploymentStatus(allRSs, newRS, deployment)
 }
 
+func (dc *DeploymentController) waitForSyncedReplicaSets(deployment *extensions.Deployment) error {
```

Comment:
Suggest name this to something like `waitForReplicaSetsInactive` and add some comments